mllib/src/test/scala/org/apache/spark/ml/feature/DiscreteCosineTransformerSuite.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.feature
+
+import scala.beans.BeanInfo
+import scala.collection.mutable.ArrayBuffer
+
+import edu.emory.mathcs.jtransforms.dct.DoubleDCT_1D
```

Comment:
https://github.com/wendykierp/JTransforms/blob/master/src/main/java/org/jtransforms/dct/DoubleDCT_1D.java#L201 shows that jTransforms DCT defers non 2^N length inputs to FFT, which uses a chirp z-transform or mixed radix transform (https://github.com/wendykierp/JTransforms/blob/master/src/main/java/org/jtransforms/fft/DoubleFFT_1D.java#L119).

Therefore, I think that no padding is done.

Will document in scaladoc.